flink-python/pyflink/datastream/connectors.py

```diff
@@ -1449,6 +1455,314 @@ def build(self) -> 'PulsarSource':
+        return self
+
+    def set_producer_name(self, producer_name: str) -> 'PulsarSinkBuilder':
+        """
+        The producer name is informative, and it can be used to identify a particular producer
+        instance from the topic stats.
+        """
+        self._j_pulsar_sink_builder.setProducerName(producer_name)
+        return self
+
+    def set_topics(self, topics: Union[str, List[str]]) -> 'PulsarSinkBuilder':
+        """
+        Set a pulsar topic list for flink sink. Some topic may not exist currently, write to this
+        non-existed topic wouldn't throw any exception.
+        """
+        if not isinstance(topics, list):
+            topics = [topics]
+        self._j_pulsar_sink_builder.setTopics(topics)
+        return self
+
+    def set_delivery_guarantee(self, delivery_guarantee: DeliveryGuarantee) -> 'PulsarSinkBuilder':
+        """
+        Sets the wanted the DeliveryGuarantee. The default delivery guarantee is
+        DeliveryGuarantee#NONE.
+        """
+        self._j_pulsar_sink_builder.setDeliveryGuarantee(
+            delivery_guarantee._to_j_delivery_guarantee())
+        return self
+
+    def set_topic_routing_mode(self, topic_routing_mode: TopicRoutingMode) -> 'PulsarSinkBuilder':
+        """
+        Set a routing mode for choosing right topic partition to send messages.
+        """
+        self._j_pulsar_sink_builder.setTopicRoutingMode(
+            topic_routing_mode._to_j_topic_routing_mode())
+        return self
+
+    def set_serialization_schema(self, pulsar_serialization_schema: PulsarSerializationSchema) \
+            -> 'PulsarSinkBuilder':
+        """
+        Sets the PulsarSerializationSchema that transforms incoming records to bytes.
+        """
+        self._j_pulsar_sink_builder.setSerializationSchema(
+            pulsar_serialization_schema._j_pulsar_serialization_schema)
+        return self
+
+    def enable_schema_evolution(self) \
+            -> 'PulsarSinkBuilder':
+        """
+        If you enable this option, we would consume and deserialize the message by using Pulsar
+        Schema.
+        """
+        self._j_pulsar_sink_builder.enableSchemaEvolution()
+        return self
+
+    def delay_sending_message(self, message_delayer: MessageDelayer):
+        """
+        Set a message delayer for enable Pulsar message delay delivery.
+        """
+        self._j_pulsar_sink_builder.delaySendingMessage(message_delayer._j_message_delayer)
+        return self
+
+    def set_config(self, key: ConfigOption, value) -> 'PulsarSinkBuilder':
+        """
+        Set an arbitrary property for the PulsarSink and Pulsar Producer. The valid keys can be
+        found in PulsarSinkOptions and PulsarOptions.
+
+        Make sure the option could be set only once or with same value.
+        """
+        self._j_pulsar_sink_builder.setConfig(key._j_config_option, value)
+        return self
+
+    def set_config_with_dict(self, config: Dict) -> 'PulsarSinkBuilder':
```

Comment:
```suggestion
    def set_properties(self, config: Dict) -> 'PulsarSinkBuilder':
```
What about rename it to set_properties to align it with setProperties?

Reply:
I change only the method name. Same as before, I call setConfig(JConfiguration.fromMap(config)) in this method. Because Python dict can auto convert Java HashMap, using setProperties is not convenient.